### PR TITLE
Added support for M2 and M5

### PR DESCRIPTION
**Pull request by mrgunior**
I currently hardcoded the TENANT provider in the catalog. This means users are able to see M2 and M5 plan sizes under TENANT in the marketplace. Apart from that just a simple check in instance_operations.go. Users are now able to create M2 or M5 clusters. 

These are the only parameters users have access to for M2 and M5 clusters:
```
"providerSettings": {
   "providerName": "TENANT",
   "backingProviderName": "AWS",
   "instanceSizeName": "M2",
   "regionName": "EU_WEST_1"
}
```

The only 3 parameters that can be changed here are:
- backingProviderName
- regionName
- instanceSizeName

Everything else are required to keep their current value as specified by the [docs](https://docs.atlas.mongodb.com/reference/api/clusters-create-one/)

### CODE REVIEW

**Comment by mrgunior** [comment]
> I think I would rather use `TENANT`, to be consistent with the API.
> 
> What happens if someone specifies parameters that aren't included in those three?

M2 and M5 clusters are very strict.
you can't wiggle yourself out and just leave parameters out, the minimum are all required and work together. Leaving any of them out will just return a failure from atlas by default. 

Error could be missing plan_id or in most cases this:
{
    "description": "atlas error: [MISSING_ATTRIBUTE] The required attribute providerName was not specified."
}

File: pkg/broker/instance_operations.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ const (
 	OperationProvision   = "provision"
 	OperationDeprovision = "deprovision"
 	OperationUpdate      = "update"
+	M2                   = "M2"
+	M5                   = "M5"
```
[the remark this answers is not in the document]
Comment (mrgunior) [reply pkg/broker/instance_operations.go]: Agreed.

**Comment by BenElgar** [comment]
I'm still not clear on what happens if a user adds parameters that aren't applicable to a service instance. Say that a user applies the following service instance:
```yaml
---
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceInstance
metadata:
  name: atlas-cluster-instance
spec:
  clusterServiceClassExternalName: mongodb-atlas-tenant
  clusterServicePlanExternalName: M2
  parameters:
    cluster:
      providerSettings:
        regionName: "EU_CENTRAL_1"
        diskTypeName: "P4"   # not applicable for the M2
        backupEnabled: true  # not applicable for the M2
```

Will they successfully provision a cluster that doesn't have those settings set? Will they get an error? If so, how will the error be displayed to the user? What happens if they specify a bad region?

**Comment by mrgunior** [comment]
> I'm still not clear on what happens if a user adds parameters that aren't applicable to a service instance. Say that a user applies the following service instance:
> 
> ```yaml
> ---
> apiVersion: servicecatalog.k8s.io/v1beta1
> kind: ServiceInstance
> metadata:
>   name: atlas-cluster-instance
> spec:
>   clusterServiceClassExternalName: mongodb-atlas-tenant
>   clusterServicePlanExternalName: M2
>   parameters:
>     cluster:
>       providerSettings:
>         regionName: "EU_CENTRAL_1"
>         diskTypeName: "P4"   # not applicable for the M2
>         backupEnabled: true  # not applicable for the M2
> ```
> 
> Will they successfully provision a cluster that doesn't have those settings set? Will they get an error? If so, how will the error be displayed to the user? What happens if they specify a bad region?

No new provision get's created. Currently what the user sees when using kubernetes is simply:
1) serviceinstance.servicecatalog.k8s.io/atlas-cluster-instance created
2) when executing this `kubectl get serviceinstances -n atlas` we get the following under status:
`ProvisionRequestInFlight`, afterwards `OrphanMitigation`, and then `OrphanMitigationSuccessful`. 

Which basically means according to this [link](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#orphan-mitigation), To mitigate orphaned Service Instances and Service Bindings, the Platform SHOULD attempt to delete resources it cannot be sure were successfully created, and SHOULD keep trying to delete them until the Service Broker responds with a success.

Now for a correct provision with all of the correct values per parameter:
1) We still get serviceinstance.servicecatalog.k8s.io/atlas-cluster-instance created
2) But now, we get the following on status: `Provisioning` as expected. If we were to head to Atlas we can see if there being provisioned.

**Pull request merged**